include/solvers/euler_solver.h

```diff
@@ -92,6 +92,21 @@ class EulerSolver : public FirstOrderUnsteadySolver
   virtual bool nonlocal_residual (bool request_jacobian,
                                   DiffContext &) override;
 
+  /**
+   * A method to integrate the system::QoI functionals.
+   */
+  virtual void integrate_qoi_timestep() override;
+
+  /**
+   * A method to compute the adjoint refinement error estimate at the current timestep.
+   * int_{tstep_start}^{tstep_end} R(u^h,z) dt
+   * Fills in an ErrorVector that contains the weighted sum of errors from all the QoIs and can be used to guide AMR.
+   * Also fills in a map that links QoI indices to spatially integrated error estimates for the QoI with that index.
+   * The trapezoidal rule (for temporally smooth QoI) or left Riemann sum (for an instantaneous QoI) is used
```

Comment:
Here, on the other hand, we shouldn't be using the trapezoidal rule for any theta; even theta=0.5 should be doing a midpoint evaluation; that's the whole distinction between EulerSolver and Euler2Solver.  It would be better to have nothing but a libmesh_not_implemented() override here, so we know to fix it if/when we need it, than to have something subtly inconsistent.

Reply:
Done.